### ISSUE TITLE
inline image support for bytes from file or network

### ISSUE THREAD

**Issue by liudonghua123** [elapouya/python-docx-template#468]
## Is your feature request related to a problem? Please describe.
I want to render a docx which contains some images from network, I find it is difficult to do it. 

I have to download and save as a temp file, use the current [InlineImage](https://docxtpl.readthedocs.io/en/latest/#inline-image) and then I have to remember to remove it.

Why not directly support image contents as input?

## Describe the solution you'd like

Support the image contents when using [InlineImage](https://docxtpl.readthedocs.io/en/latest/#inline-image).

## Describe alternatives you've considered

n.a.

## Additional context

n.a.

**Comment by cw0516** [comment]
I hope this feature to be added too👏

**Comment by radove** [comment]
Hello, Its possible to pass an io.BytesIO(bytes) to the InlineImage() function.  Perhaps there needs to be a new example leveraging this method: @cw0516 @liudonghua123

```
fig = px.scatter(x=[0, 1, 2, 3, 4], y=[0, 1, 4, 9, 16]) #plotly Image example
img_bytes = fig.to_image(format="png",width=600, height=350) # plotly image generate bytes
main_visualization = io.BytesIO(img_bytes) # convert bytes to File obj
context = { 'main_visualization': docxtpl.InlineImage(doc, main_visualization) } # python-docx-template will believe this is a File and run seek on it.
```

**Comment by liudonghua123** [comment]
> Hello, Its possible to pass an io.BytesIO(bytes) to the InlineImage() function. Perhaps there needs to be a new example leveraging this method: @cw0516 @liudonghua123
> 
> ```
> fig = px.scatter(x=[0, 1, 2, 3, 4], y=[0, 1, 4, 9, 16]) #plotly Image example
> img_bytes = fig.to_image(format="png",width=600, height=350) # plotly image generate bytes
> main_visualization = io.BytesIO(img_bytes) # convert bytes to File obj
> context = { 'main_visualization': docxtpl.InlineImage(doc, main_visualization) } # python-docx-template will believe this is a File and run seek on it.
> ```

Yeah, I found `new_pic_inline` in upstream project `docx` support `IO[bytes]`, see https://github.com/python-openxml/python-docx/blob/57d3b9ee9cb778e76258653c6dc464d5598ca80b/src/docx/parts/story.py#L60-L65.

**Comment by liudonghua123** [comment]
And I tested, it worked as expected!

```python
import io
from docxtpl import DocxTemplate, InlineImage
from docx.shared import Mm

doc = DocxTemplate("test.docx")
context = { 'company_name' : "World company", 'inline_image': InlineImage(doc, io.BytesIO(open('test.png', 'rb').read()), width=Mm(10), height=Mm(10))}
doc.render(context)
doc.save("generated_doc.docx")
```

![image](https://github.com/elapouya/python-docx-template/assets/2276718/bf7f4e45-aff0-42ef-83a0-dd0b659caeca)

**Comment by liudonghua123** [comment]
I can also use file like object as the second argument of InlineImage.

![image](https://github.com/elapouya/python-docx-template/assets/2276718/18125f64-cd93-4354-b80a-19655ac32157)
